aten/src/ATen/native/TensorAdvancedIndexing.h

```diff
@@ -39,4 +40,48 @@ DECLARE_DISPATCH(scatter_reduce_two_fn, scatter_reduce_two_stub);
 
 TORCH_API Tensor& index_out(Tensor& result, const Tensor & self, const c10::List<c10::optional<at::Tensor>>& indices);
 
+// fast paths for GNN usage
+template <bool is_scatter_like = true>
```

Comment:
It doesn't seem to me like there  is any good reason to template over this bool

Reply:
fixed!